[CI] QueryStringFunctionTests class failing

**Build Scans:**
- [elasticsearch-periodic #4186 / release-tests](https://gradle-enterprise.elastic.co/s/dusdn3ffxwb32)
- [elasticsearch-pull-request #32799 / release-tests](https://gradle-enterprise.elastic.co/s/6urtdrbijqw5o)
- [elasticsearch-periodic #4182 / release-tests](https://gradle-enterprise.elastic.co/s/wuay57ptp6czo)
- [elasticsearch-periodic #4178 / release-tests](https://gradle-enterprise.elastic.co/s/fmvqxwlkqkx3m)
- [elasticsearch-periodic #4174 / release-tests](https://gradle-enterprise.elastic.co/s/kdhilfl2k7ave)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.fulltext.QueryStringFunctionTests.testSerializationOfSimple {TestCase=type error for <geo_point>}" -Dtests.seed=E470A36DB39E4C1A -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=haw-US -Dtests.timezone=Etc/GMT-14 -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.expression.function.fulltext.QueryStringFunctionTests'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.expression.function.fulltext.QueryStringFunctionTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Unknown NamedWriteable [org.elasticsearch.xpack.esql.core.expression.Expression][QStr]
```

**Issue Reasons:**
- [main] 5 consecutive failures in step release-tests
- [main] 5 failures in class org.elasticsearch.xpack.esql.expression.function.fulltext.QueryStringFunctionTests (2.0% fail rate in 245 executions)
- [main] 5 failures in step release-tests (100.0% fail rate in 5 executions)
- [main] 4 failures in pipeline elasticsearch-periodic (100.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -287,8 +287,6 @@ tests:
 - class: org.elasticsearch.validation.DotPrefixClientYamlTestSuiteIT
   method: test {p0=dot_prefix/10_basic/Deprecated index template with a dot prefix index pattern}
   issue: https://github.com/elastic/elasticsearch/issues/113529
-- class: org.elasticsearch.xpack.esql.expression.function.fulltext.QueryStringFunctionTests
-  issue: https://github.com/elastic/elasticsearch/issues/113496
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search/180_locale_dependent_mapping/Test Index and Search locale dependent mappings / dates}
   issue: https://github.com/elastic/elasticsearch/issues/113537

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/fulltext/QueryStringFunctionTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/fulltext/QueryStringFunctionTests.java
@@ -18,16 +18,23 @@ import org.elasticsearch.xpack.esql.expression.function.AbstractFunctionTestCase
 import org.elasticsearch.xpack.esql.expression.function.FunctionName;
 import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
 import org.hamcrest.Matcher;
+import org.junit.BeforeClass;
 
 import java.util.Arrays;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.function.Supplier;
 
+import static org.elasticsearch.xpack.esql.action.EsqlCapabilities.Cap.QSTR_FUNCTION;
 import static org.hamcrest.Matchers.equalTo;
 
 @FunctionName("qstr")
 public class QueryStringFunctionTests extends AbstractFunctionTestCase {
+
+    @BeforeClass
+    public static void checkFunctionEnabled() {
+        assumeTrue("QSTR capability should be enabled ", QSTR_FUNCTION.isEnabled());
+    }
 
     public QueryStringFunctionTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
         this.testCase = testCaseSupplier.get();

